packages/components/nodes/memory/DynamoDb/DynamoDb.ts

```diff
@@ -142,18 +142,26 @@ interface DynamoDBSerializedChatMessage {
 }
 
 class BufferMemoryExtended extends FlowiseMemory implements MemoryMethods {
+    tableName = ''
     sessionId = ''
     dynamodbClient: DynamoDBClient
+    dynamoKey = ''
+    partitionKey = ''
 
     constructor(fields: BufferMemoryInput & BufferMemoryExtendedInput) {
         super(fields)
         this.sessionId = fields.sessionId
         this.dynamodbClient = fields.dynamodbClient
+
+        // These fields are coming in on chatHistory, but should they be on the dynamodbClient instead?
+        this.partitionKey = (fields?.chatHistory as unknown as { partitionKey: string }).partitionKey
```

Comment:
I'm not sure why its coming from `chatHistory`, I think a better way is to pass through the initilization of `BufferMemoryExtended` at line 121?

Reply:
Thanks for pointing that out. I have b better understanding now. I'll update the PR...

Reply:
During the init, chatHistory is assigned the value of dynamoDb, which is the result of DynamoDBChatMessageHistory. That's the traceback for why fields.chatHistory has all those attributes. Placed the pass through on the initialization of BufferMemoryExtended. Code is working as expected and much easier to follow.

Reply:
there are other places we are still using `(this as any).dynamoKey` or `partitionKey` or `tableName`, can we just do `this` as well?

Reply:
You mean change it back to (this as any) instead of adding them as variables to the class? I had added them as variables to the class to avoid using (this as any), but I should have added them as private since they really aren't used outside the class.

Reply:
No I meant we probably can get rid of using `this as any` and just use `this` instead

Reply:
Oh, man! I was tired when I was reading that comment last night. I see what you mean.... Yes, I'll update in other uses! 😀

Reply:
@HenryHengZJ replaced all (this as any) usage in BufferMemoryExtended. Tested using session and override session again, and all chat threads continue to produce the expected results.

Reply:
tysm!